web-ng/src/app/components/drawing-tools/drawing-tools.component.ts

```diff
@@ -14,48 +14,53 @@
  * limitations under the License.
  */
 
-import { Component, ChangeDetectionStrategy } from '@angular/core';
+import {
+  Component,
+  ChangeDetectionStrategy,
+  Input,
+  OnChanges,
+} from '@angular/core';
 import {
   DrawingToolsService,
   EditMode,
 } from '../../services/drawing-tools/drawing-tools.service';
-import { ProjectService } from '../../services/project/project.service';
-import { Observable } from 'rxjs';
+import { Project } from './../../shared/models/project.model';
 import { Layer } from '../../shared/models/layer.model';
 import { List } from 'immutable';
-import { map } from 'rxjs/internal/operators/map';
 import { getPinImageSource } from '../map/ground-pin';
 import { DomSanitizer, SafeUrl } from '@angular/platform-browser';
 
 @Component({
-  selector: 'drawing-tools',
+  selector: 'ground-drawing-tools',
   templateUrl: './drawing-tools.component.html',
   styleUrls: ['./drawing-tools.component.css'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class DrawingToolsComponent {
+export class DrawingToolsComponent implements OnChanges {
+  @Input() project!: Project;
+
   pointValue = 'point';
   polygonValue = 'polygon';
   selectedValue = '';
   private lastSelectedValue = '';
   selectedLayerId = '';
-  readonly layers$: Observable<List<Layer>>;
+  layers!: List<Layer>;
   readonly lang: string;
 
   constructor(
     private drawingToolsService: DrawingToolsService,
-    private sanitizer: DomSanitizer,
-    projectService: ProjectService
+    private sanitizer: DomSanitizer
   ) {
     // TODO: Make dynamic to support i18n.
     this.lang = 'en';
-    this.layers$ = projectService.getActiveProject$().pipe(
-      map(project => {
-        this.selectedLayerId = project.layers.keySeq().first();
-        this.drawingToolsService.setSelectedLayerId(this.selectedLayerId);
-        return List(project.layers.valueSeq().toArray()).sortBy(l => l.index);
-      })
-    );
+  }
+
+  ngOnChanges(): void {
```

Comment:
@DaoyuT @parulraheja98 @gauravchalana1 @gyarill I tried using `@Input` and `ngOnChanges` to simplifying making the application reactive to changes in the project config, but this approach makes me sad. The reason it makes me sad is that on some cases we still need the activeProject$ stream (e.g, when we're merging with feature or observation streams), and in other we don't want to react to changes in the project config (e.g., when editing an observation). This means we get up with at least three different ways of accessing the active project.

I personally prefer consistency and not needing to decide each time which approach to use, so considering rolling back to pure RxJS (no `@Input` properties, no cached `activeProject` in `projectService`).

Before I do this, do you have any thoughts/ideas/opinions about this?